risc0/zkvm/src/host/client/env.rs

```diff
@@ -77,6 +77,12 @@ pub struct ProveZkrRequest {
     pub input: Vec<u8>,
 }
 
+/// Wrapper newtype to enforce alignment to allow casting to KeccakState, which is u64 aligned
+/// (8 bytes).
+#[repr(align(8))]
+#[derive(Clone)]
+pub struct KeccakInput(pub Vec<u8>);
```

Comment:
I'm pretty sure that this `repr(align(8))` here doesn't do what you want. This will affect the alignment of the memory that the `Vec` is in (which is just a size and pointer) but not the buffer that the `Vec` allocates, which IIUC is what is needed.

It seems like we need to somehow either:
1. change the type of the Vec before it allocates memory
2. copy the memory to somewhere that is aligned properly before casting

Reply:
might just be easiest to just change the `Vec<u8>` to a `Vec<u64>`

Reply:
resolved in below commits (sorry for not having one commit to fix, was trying to avoid installing latest toolchain)